android/src/com/unciv/app/NativeFontAndroid.kt

```diff
@@ -6,20 +6,42 @@ import android.graphics.Paint
 import android.graphics.Typeface
 import android.graphics.fonts.Font
 import android.graphics.fonts.FontFamily
+import android.graphics.fonts.FontStyle
 import android.graphics.fonts.SystemFonts
 import android.os.Build
 import com.badlogic.gdx.graphics.Pixmap
-import com.unciv.ui.utils.FontData
+import com.unciv.ui.utils.FontFamilyData
 import com.unciv.ui.utils.NativeFontImplementation
+import java.util.*
+import kotlin.math.abs
 
 /**
  * Created by tian on 2016/10/2.
  */
-class NativeFontAndroid(private val size: Int, private val fontFamily: String) :
-    NativeFontImplementation {
+class NativeFontAndroid(
+    private val size: Int,
+    private val fontFamily: String
+) : NativeFontImplementation {
+    private val fontList =
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) emptySet()
+        else SystemFonts.getAvailableFonts()
+
     private val paint = Paint().apply {
         typeface = if (fontFamily.isNotBlank() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            val font = fontList.firstOrNull { it.file?.nameWithoutExtension == fontFamily }
+            // Helper within the VERSION_CODES.Q gate: Evaluate a Font's desirability (lower = better) for a given family. 
+            fun Font.matchesFamily(family: String): Int {
+                val name = file?.nameWithoutExtension ?: return Int.MAX_VALUE
+                if (name == family) return 0
+                if (!name.startsWith("$family-")) return Int.MAX_VALUE
+                if (style.weight == FontStyle.FONT_WEIGHT_NORMAL && style.slant == FontStyle.FONT_SLANT_UPRIGHT) return 1
+                return 2 +
+                        abs(style.weight - FontStyle.FONT_WEIGHT_NORMAL) / 100 +
+                        abs(style.slant - FontStyle.FONT_SLANT_UPRIGHT)
+            }
+            val font = fontList.mapNotNull {
+                val q = it.matchesFamily(fontFamily)
```

Comment:
Rename q to something more significant

Reply:
Of course, sounds like an interim never meant to stay. Still busy on another branch that like this one started from a "that's easy" trigger and got way complicated really fast... Soon.